pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,338 @@
+		return
+	}
+	for _, info := range infoList {
+		_, err := fmt.Fprintf(writer, formatStr, info.Name, info.ReadyContainers, info.Reason, info.Restarts, info.CreationTimestamp)
+		if err != nil {
+			fmt.Printf("Failed to print Pod information: %s.\n", err)
+			return
+		}
+	}
+}
+
+// filterPods filters pods based on the provided filter callback function.
+func filterPods(pods *corev1.PodList, filterFunc func(job corev1.Pod) bool) *corev1.PodList {
+	filteredPods := &corev1.PodList{}
+	for _, pod := range pods.Items {
+		if filterFunc(pod) {
+			filteredPods.Items = append(filteredPods.Items, pod)
+		}
+	}
+	return filteredPods
+}
+
+// createLabelSelector creates a label selector based on the provided job name or queue name.
+func createLabelSelector(jobName, queueName string) (labels.Selector, error) {
+	var labelSelector labels.Selector
+	reqs := []labels.Requirement{}
+
+	// Add label requirements based on jobName label
+	if jobName != "" {
+		inRequirement, err := labels.NewRequirement(v1alpha1.JobNameKey, selection.In, []string{jobName})
+		if err != nil {
+			return nil, err
+		}
+		reqs = append(reqs, *inRequirement)
+	}
+	// Add label requirements based on queueName label
+	if queueName != "" {
+		inRequirement, err := labels.NewRequirement(v1alpha1.QueueNameKey, selection.In, []string{queueName})
+		if err != nil {
+			return nil, err
+		}
+		reqs = append(reqs, *inRequirement)
+	}
+
+	if len(reqs) > 0 {
+		// If job name or queue name are provided, select pods with both labels.
+		labelSelector = labels.NewSelector().Add(reqs...)
+	} else {
+		// If job name is not provided, select all pods created by vcjobs.
+		inRequirement, err := labels.NewRequirement(v1alpha1.JobNameKey, selection.Exists, []string{})
+		if err != nil {
+			return nil, err
+		}
+		labelSelector = labels.NewSelector().Add(*inRequirement)
+	}
+	return labelSelector, nil
+}
+
+// translateTimestampSince translates a timestamp into a human-readable string using time.Since.
+func translateTimestampSince(timestamp metav1.Time) string {
+	if timestamp.IsZero() {
+		return "<unknown>"
+	}
+	return duration.HumanDuration(time.Since(timestamp.Time))
+}
+
+// PodInfo holds information about a pod.
+type PodInfo struct {
+	Name              string
+	ReadyContainers   string
+	Reason            string
+	Restarts          string
+	CreationTimestamp string
+}
+
+// Lengths holds the maximum length of each column.
+type Lengths struct {
+	Name              int
+	ReadyContainers   int
+	Reason            int
+	Restarts          int
+	CreationTimestamp int
+}
+
+// printPod information in a tabular format.
+func printPod(pod *corev1.Pod) (PodInfo, Lengths) {
+	restarts := 0
+	totalContainers := len(pod.Spec.Containers)
+	readyContainers := 0
+	lastRestartDate := metav1.NewTime(time.Time{})
+
+	reason := string(pod.Status.Phase)
+	if pod.Status.Reason != "" {
+		reason = pod.Status.Reason
+	}
+
+	initializing := false
+	for i := range pod.Status.InitContainerStatuses {
```

Comment:
Here should change when this feature https://github.com/volcano-sh/volcano/issues/3513 is merged, you can add a TODO and add this link here: )

Reply:
I have considered initContainer do I still need to document this TODO?